platform-service/src/unstract/platform_service/helper/cost_calculation.py

```diff
@@ -83,18 +133,31 @@ def _fetch_and_save_json(self) -> Optional[dict[str, Any]]:
             response.raise_for_status()
             json_data = response.json()
             # Save JSON data to file
-            with open(self.file_path, "w", encoding="utf-8") as f:
-                json.dump(json_data, f, ensure_ascii=False, indent=4)
-            # Set the file's modification time to indicate TTL
-            expiry_date = datetime.now() + timedelta(days=self.ttl_days)
-            expiry_timestamp = expiry_date.timestamp()
-            os.utime(self.file_path, (expiry_timestamp, expiry_timestamp))
+            if check_feature_flag_status(FeatureFlag.REMOTE_FILE_STORAGE):
+                self.file_storage.json_dump(
+                    path=self.file_path,
+                    mode="w",
+                    encoding="utf-8",
+                    data=json_data,
+                    ensure_ascii=False,
+                    indent=4,
+                )
+            else:
+                with open(self.file_path, "w", encoding="utf-8") as f:
+                    json.dump(json_data, f, ensure_ascii=False, indent=4)
+                # Set the file's modification time to indicate TTL
+                expiry_date = datetime.now() + timedelta(days=self.ttl_days)
+                expiry_timestamp = expiry_date.timestamp()
+                os.utime(self.file_path, (expiry_timestamp, expiry_timestamp))
```

Comment:
@gaya3-zipstack are we intentionally not updating the modification time in case of remote storage?

Reply:
Yes. Because whenever we write to a file, the modification time will reflect the last time that the file is written to. There is no need to explicitly update the time in this particular use-case.